core_backend/app/db/db_models.py

```diff
@@ -316,71 +321,258 @@ class ContentDB(Base):
     SQL Alchemy data model for content
     """
 
-    __tablename__ = "content"
+    __tablename__ = "contents"
 
     content_id: Mapped[int] = mapped_column(Integer, primary_key=True, nullable=False)
+
+    def __repr__(self) -> str:
+        """Pretty Print"""
+        return f"<Content #{self.content_id}>"
+
+
+class LanguageDB(Base):
+    """
+    SQL Alchemy data model for language
+    """
+
+    __tablename__ = "languages"
+
+    language_id: Mapped[int] = mapped_column(Integer, primary_key=True, nullable=False)
+
+    language_name: Mapped[str] = mapped_column(String, nullable=False)
+    is_default: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
+    created_datetime_utc: Mapped[datetime] = mapped_column(DateTime, nullable=False)
+    updated_datetime_utc: Mapped[datetime] = mapped_column(DateTime, nullable=False)
+
+    __table_args__ = (
+        Index(
+            "ix_languages_is_default_true",
+            "is_default",
+            unique=True,
+            postgresql_where=(is_default.is_(True)),
+        ),
+    )
+
+    def __repr__(self) -> str:
+        """Pretty Print"""
+        return f"<Language #{self.language_id}:  {self.language_name}>"
+
+
+async def save_language_to_db(
+    language: LanguageBase,
+    asession: AsyncSession,
+) -> LanguageDB:
+    """
+    Saves a new language in the database
+    """
+
+    language_db = LanguageDB(
+        language_name=language.language_name,
+        is_default=language.is_default,
+        created_datetime_utc=datetime.utcnow(),
+        updated_datetime_utc=datetime.utcnow(),
+    )
+    asession.add(language_db)
+
+    await asession.commit()
+    await asession.refresh(language_db)
+
+    return language_db
+
+
+async def update_language_in_db(
+    language_id: int,
+    language: LanguageBase,
+    asession: AsyncSession,
+) -> LanguageDB:
+    """
+    Updates a language in the database
+    """
+
+    language_db = LanguageDB(
+        language_id=language_id,
+        is_default=language.is_default,
+        language_name=language.language_name,
+        updated_datetime_utc=datetime.utcnow(),
```

Comment:
Does this not need `created_datetime_utc=language.created_datetime_utc`?

Reply:
Since we are trying to update, not adding it won't affect the DB.